### DISCUSSION TITLE
How Can I Optimize my NextJS _app.js build file?

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [vercel/next.js#41373] [Help]
I am trying to optimize my portfolio website built using NextJS. The problem is mainly performance. According to the lighthouse score :

- Performance - 57
- SEO - 100
- Accessibility - 97
- Best Practices - 100

I'm aware of the advanced optimizing techniques & hence applied all of the best practices like _lazy-loading_ the images and third-party libraries, setting `prefetch` to `false` to the links, using `getStaticPaths` + `getStaticProps` for requesting the content from CMS using `graphql-request`, minimizing the CSS, even using `React.Memo` for Pure components.

Even after doing all of that, my Performance is stuck at 57 and it does take a while to render the first page on the screen.

According to the lighthouse, the main problem being, "Avoid enormous network payloads". Also I can see my performance culprits such as `_app.js` bundle to be 2245kb & `chunks/main` bundle to be 1128kb.

I'm not being able to reduce these bundle size even after all the mentioned techniques.

Here's the link where I have hosted my site - [abhi.at](https://abhi.at/) & Github - https://github.com/abhinav-anshul/abhi.at

If anyone who would point me in the right direction would be very helpful. Thanks so much for reading!

**Comment by icyJoseph** [comment]
Hi,

I used a 12 times slow down multiplier on my CPU and got 96-98 performance scores on your site. Perhaps you've done some fixes since posting this?

Here's a [breakdown of your bundles](https://googlechrome.github.io/lighthouse/treemap/?gzip=1#H4sIAAAAAAAAE62WyW7bMBCGX8XQ2ZK4LzkWQU/JoU2MHooi4GqzkSVVousagd+9tJ2TygSo4qM0yzfzazjiS9FshuLmpRjcr50bo7OroSluik2M/XhT1/v9vlJ6EyoV62JZ+NCq5l0PtbMhjqeMoxlCH8s4OLdVfWlVVKfXwabgnG1ZxBAbl6wPZ+vi8WJd3F6s1l2iQtcmn9Xo7MJ3w+I1x0L1fXIaTTekFO2uaV4fbsPYN+pw39lT6tCmmK2K4bc7p4wqNOdq46E/2ScVtSkqmb+/FK3aure7HtzY7QbjPh3iyZ8jzshx+W7UU+v+xHqMqRZTm82ufR7rvmsOPjTNWBrGFacWaualtEZUP8d/MWAGQkBSCcy5BsgzrJwGQGeTQ4aWxa7dJZ0/gGOMVEwwbARGFmMshOJZHBV0yoMQzxFRVJZQrpEA1lDDGFL5/gC5Sn+y8gZ5raEwVHIsAczikGBgwkMSzyAiKSvBUywnQhPgrSN5YqrkCg0SnhpM34Z54gyw1iuG8jjIr4BDiFeMeeicx1gKyRlleT0Rn34/RMQM4t7pXpnnEnoKCOZKCu0kM3kqxYBOp1QSNgPrh+S/74bnklDKDKHWpnnXQuL8tBIg8XSACBEUzmBvVWhLbZR3wEHpFONYuzcOCeNgei4pAHSO0r1au7F+Sou6JBY57IWECGhlaH69MSnAtGXIKEBz5orAkiDimUibSBoljM6PcVpYeHpuEGJgjs6XhkNr3Z/Se4iRSusPI+0x91m4QERO2PK/lF6XuPy8/3b4sj4QcnfXPH4tV/f1k96Fxt6rNvj0i8+TyYe24RvccVy/S73ODr7IrJsuVWEBhJQr4RDBkuaHmhI4VZlQyI8/jsfjsjBd68P6wcUY2vX5atB0Rp3vJa4tVw9F8voLdO9TqTYJAAA=), automatically made by lighthouse, where your `_app` bundle is about 62kb.

I also run coverage tests and at worst, you have like 20% unused JS on first load, but that's framework stuff mostly. The blog page is also quite alright.

You should run lighthouse at least in Incognito mode and at best, from the terminal using a slow down multiplier appropriate for your CPU: https://lighthouse-cpu-throttling-calculator.vercel.app/

**Comment by abhinav-anshul** [comment]
Hi,

Thank you very much for all the efforts you took while answering this.
I do get a very good lighthouse score while running the abhi.at 

But when I run it in localhost (still in incognito mode), the performance gets struck to 57 according to the lighthouse.  I did a similar Webtestpage.org (for abhi.at) and it doesn't look quite promising. There are a few metrics that take more than 1 sec to load.


Is this the most optimization I get on this one? Or there are a few practices that I am missing from my codebase?
